pkg/command/discovery_source.go

```diff
@@ -188,3 +195,21 @@ func checkDiscoverySource(source configtypes.PluginDiscovery) error {
 	_, err = discObject.List()
 	return err
 }
+
+// ====================================
+// Shell completion functions
+// ====================================
+func completeDiscoverySources(_ *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) > 0 {
```

Comment:
nit (because it is unlikely the list of sources is long, but): for consistency should we sort this list as well?

Reply:
At the moment, the CLI only supports a single plugin source on which we can use `tanzu plugin source delete/update`.  All other plugin sources are set using `TANZU_CLI_ADDITIONAL_PLUGIN_DISCOVERY_IMAGES_TEST_ONLY`, and the `tanzu plugin source delete/update` commands don't operate on them.

That being said, I'm still going to do sorting as you suggest as it will make use more future-proof for when we support 3rd-party repositories.